plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/service/CodeWhispererService.kt

```diff
@@ -94,6 +96,10 @@ import java.util.concurrent.TimeUnit
 class CodeWhispererService(private val cs: CoroutineScope) : Disposable {
     private val codeInsightSettingsFacade = CodeInsightsSettingsFacade()
     private var refreshFailure: Int = 0
+    private val ongoingRequests = mutableMapOf<Int, InvocationContext?>()
+    val ongoingRequestsContext = mutableMapOf<Int, RequestContext>()
```

Comment:
not sure if this is the best pattern if the exact jobId is only relevant for "which is the last request", which we can determine in other ways

Reply:
jobId is only to separate different triggers and is a local identifier only, (same as sessionId except we know the jobId as soon as a trigger happens before receiving the response)